x-pack/plugins/apm/public/components/app/ServiceDetails/ServiceIntegrations/MachineLearningFlyout.tsx

```diff
@@ -0,0 +1,256 @@
+    if (
+      prevProps.serviceName !== this.props.serviceName ||
+      prevProps.transactionType !== this.props.transactionType
+    ) {
+      this.checkForMlJob();
+    }
+  }
+
+  public async checkForMlJob() {
+    const { serviceName, transactionType } = this.props;
+    const { count } = await getMlJob({
+      serviceName,
+      transactionType
+    });
+    this.setState({ hasMlJob: count > 0 });
+  }
+
+  public createJob = async () => {
+    this.setState({ isLoading: true });
+    try {
+      const { serviceName, transactionType } = this.props;
+      if (serviceName) {
+        const res = await startMlJob({ serviceName, transactionType });
+        const didSucceed = res.datafeeds[0].success && res.jobs[0].success;
+        if (!didSucceed) {
+          throw new Error('Creating ML job failed');
+        }
+        this.addSuccessToast();
+      }
+    } catch (e) {
+      this.addErrorToast();
+    }
+
+    this.setState({ isLoading: false });
+    this.props.onClose();
+  };
+
+  public addErrorToast = () => {
+    const {
+      serviceName,
+      transactionType = 'all transactions',
+      location
+    } = this.props;
+    toastNotifications.addWarning({
+      title: 'Job already exists',
+      text: (
+        <p>
+          There&apos;s already a job running for anomaly detection on{' '}
+          {serviceName} ({transactionType}
+          ).{' '}
+          <ViewMLJob
+            serviceName={serviceName}
+            transactionType={transactionType}
+            location={location}
+          >
+            View existing job
+          </ViewMLJob>
+        </p>
+      )
+    });
+  };
+
+  public addSuccessToast = () => {
+    const {
+      serviceName,
+      transactionType = 'all transactions',
+      location
+    } = this.props;
+    toastNotifications.addSuccess({
+      title: 'Job successfully created',
+      text: (
+        <p>
+          The analysis is now running for {serviceName} ({transactionType}
+          ). It might take a while before results are added to the response
+          times graph.{' '}
+          <ViewMLJob
+            serviceName={serviceName}
+            transactionType={transactionType}
+            location={location}
+          >
+            View job
+          </ViewMLJob>
+        </p>
+      )
+    });
+  };
+
+  public render() {
+    const { isOpen, onClose, serviceName, transactionType } = this.props;
+    const { isLoading, hasIndexPattern, hasMlJob } = this.state;
+
+    if (!isOpen) {
+      return null;
+    }
+
+    return (
+      <EuiFlyout onClose={onClose} size="s">
+        <EuiFlyoutHeader>
+          <EuiTitle>
+            <h2>Enable anomaly detection on response times</h2>
+          </EuiTitle>
+          <EuiSpacer size="s" />
+        </EuiFlyoutHeader>
+        <EuiFlyoutBody>
+          {hasMlJob && (
+            <div>
+              <EuiCallOut
+                title={<span>Job already exists</span>}
```

Comment:
Why is the `<span>` necessary?

Reply:
I think it's required by EUI. Would be nice if simple `string` types were also allowed.

Reply:
In the EUI docs, the propType is listed as `node` and the example code only shows a string being passed in for `title`

Reply:
Okay, maybe I'm wrong. I've just previously experienced the need for wrapping in `Fragment` et al.

Reply:
EuiTitle and possibly EuiText have some sort of requirement along these lines, but this one may not. This component was copied/pasted over so I'm not sure why the span was there originally.